central/reports/common/report_formatter.go

```diff
@@ -124,72 +115,29 @@ func Format(deployedImagesResults []DeployedImagesResult, watchedImagesResults [
 		}
 	}
 
-	numWatchedImageCVEs := 0
-	for _, r := range watchedImagesResults {
-		for _, i := range r.Images {
-			for _, c := range i.getComponents() {
-				for _, v := range c.getVulnerabilities() {
-					numWatchedImageCVEs++
-					discoveredTs := "Not Available"
-					if v.DiscoveredAtImage != nil {
-						discoveredTs = v.DiscoveredAtImage.Time.Format("January 02, 2006")
-					}
-					csvWriter.AddValue(csv.Value{
-						"",
-						"",
-						"",
-						i.Name.FullName,
-						c.Name,
-						v.Cve,
-						strconv.FormatBool(v.IsFixable),
-						v.FixedByVersion,
-						strings.ToTitle(stringutils.GetUpTo(v.Severity, "_")),
-						strconv.FormatFloat(v.Cvss, 'f', 2, 64),
-						discoveredTs,
-						v.Link,
-					})
-				}
-			}
-		}
-	}
+	empty := csvWriter.IsEmpty()
 
 	var buf bytes.Buffer
 	err := csvWriter.WriteBytes(&buf)
 	if err != nil {
-		return nil, errors.Wrap(err, "error creating csv report")
+		return nil, true, errors.Wrap(err, "error creating csv report")
```

Comment:
Hmm the boolean here indicates whether the report is empty or not. In case of error, it is empty hence returns true. But I can return false too since the caller will handle the error. Or I can redefine the boolean to be true if report has rows and false if it doesn't. That way its value will always match its definition.